### DISCUSSION TITLE
Kafka Agent - Payload

### DISCUSSION THREAD

**Discussion by pgianf** [open-telemetry/opentelemetry-java-instrumentation#9681] [Q&A] [answered]
Hello,
I'm trying usig open telemetry  agent to have automatic instrumentation of my application, I can run my microservices (Spring Boot applications) that exchange messages via Kafka along with the opentelemetry-javaagent.jar, I've used Jaeger and Grafana to analyze the trace, I would like to have more informations on the single span, so not only client_id, destination_name, payload_size_bytes, .... but having even the payload content would be really helpful during problem investigations.
So I would like to know:
1. Is it possible to have the kafka  message payload as attribute?
2. If I have to extend the Kafka instrumentation how can I proceed? The project is kafka-clients-0.11? Is it better to make a copy of it in a sibling package, disable it and then modify it? I'm having a look to the examples folder, but is there any guide or a full example?
3. Is there any documentation to understand the internals of the code, how it works, .....

thanks
Paolo

**Comment by pgianf** [comment]
Related to the point 3 I've found the documentation inside the source code. :-)
If someone could help with points 1 e 2 thanks a lot!!

**Reply by laurit** [reply, marked as answer]
1. no it is not.
2. making a copy and disabling the original should be fine. there are no examples for this.

**Reply by pgianf** [reply]
ok thank you

**Reply by laurit** [reply]
Here is a sample how you could disable the existing instrumentation from and extension https://github.com/signalfx/splunk-otel-java/blob/f778440f016640e88131d594331f0bbcaaa2cdb4/custom/src/main/java/com/splunk/opentelemetry/SplunkConfiguration.java#L188
